### DISCUSSION TITLE
源码中使用了很多 string & {} 类型，请教一下这个类型代表什么含义？

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#1292] [Q&A 2.x] [answered]
没看懂 `string & {}` 这个类型是什么意思啊，按道理来讲 `string` 跟 `{}` 没有交集吧？

**Comment by janryWang** [comment, marked as answer]
这是个类型魔法，正常我们写"aa" | "bb"的话，类型只接受aa bb，其他会报错，但是加上& {}，就可以支持已有字符串枚举的同时支持扩展字符串

**Reply by liuweiGL** [reply]
请问大佬在哪看到的啊，我没有查到相关的资料。

**Reply by liuweiGL** [reply]
能麻烦提供一下关键字吗？谢谢大佬~

**Comment by liuweiGL** [comment]
翻到了：microsoft/TypeScript#29729
